src/api/app/services/trigger_controller_service/scm_extractor.rb

```diff
@@ -30,51 +30,51 @@ def call
     private
 
     def allowed_github_event_and_action?
-      @event == 'pull_request' && @payload['action'].in?(ALLOWED_GITHUB_ACTIONS)
+      @event == 'pull_request' && @payload[:action].in?(ALLOWED_GITHUB_ACTIONS)
     end
 
     def allowed_gitlab_event_and_action?
-      @event == 'Merge Request Hook' && @payload['object_attributes']['action'].in?(ALLOWED_GITLAB_ACTIONS)
+      @event == 'Merge Request Hook' && @payload[:object_attributes][:action].in?(ALLOWED_GITLAB_ACTIONS)
     end
 
     def github_extractor_payload
-      host = URI.parse(@payload.dig('sender', 'url')).host
+      host = URI.parse(@payload.dig(:sender, :url)).host
       api_endpoint = if host.start_with?('api.github.com')
                        "https://#{host}"
                      else
                        "https://#{host}/api/v3/"
                      end
       {
         scm: 'github',
-        commit_sha: @payload.dig('pull_request', 'head', 'sha'),
-        pr_number: @payload['number'],
-        source_branch: @payload.dig('pull_request', 'head', 'ref'),
-        target_branch: @payload.dig('pull_request', 'base', 'ref'),
-        action: @payload['action'], # TODO: Names may differ, maybe we need to find our own naming (defer to service?)
-        source_repository_full_name: @payload.dig('pull_request', 'head', 'repo', 'full_name'),
-        target_repository_full_name: @payload.dig('pull_request', 'base', 'repo', 'full_name'),
+        commit_sha: @payload.dig(:pull_request, :head, :sha),
+        pr_number: @payload[:number],
+        source_branch: @payload.dig(:pull_request, :head, :ref),
+        target_branch: @payload.dig(:pull_request, :base, :ref),
+        action: @payload[:action], # TODO: Names may differ, maybe we need to find our own naming (defer to service?)
+        source_repository_full_name: @payload.dig(:pull_request, :head, :repo, :full_name),
+        target_repository_full_name: @payload.dig(:pull_request, :base, :repo, :full_name),
         event: @event,
         api_endpoint: api_endpoint
-      }.with_indifferent_access
+      }.deep_symbolize_keys
```

Comment:
All hash keys are already symbols here and keys of any nested hash possibly contained in `@payload` was already converted above in the `initialize` with `@payload = payload.deep_symbolize_keys`.

So we can safely remove `deep_symbolize_keys`:
```suggestion
      }
```

Reply:
Done.